Sources/FluentKit/Schema/DatabaseSchema.swift

```diff
@@ -81,12 +81,22 @@ public struct DatabaseSchema {
     public enum FieldConstraint {
         case required
         case identifier(auto: Bool)
+        case foreignKey(parentTable: String, parentField: FieldName, updateAction: Constraint.ForeignKeyAction, deleteAction: Constraint.ForeignKeyAction)
```

Comment:
Maybe something like this would be better:

```swift
public enum FieldConstraint {
    ...
    case reference(field: FieldName, onUpdate: ForeignKeyAction, onDelete: ForeignKeyAction)
}
```

```swift
public enum FieldName {
    ...
    case foreign(name: String, table: String)
}
```

Reply:
oh, that's very nice. thanks.

Reply:
although, unfortunately it makes the table constraint below a bit awkward as-is because the table constraint needs to hold exactly 1 parent table name but any number of parent field names. Making this change to `FieldName` would probably make `FieldName` not an ideal fit for use with the table constraint.

Then again, there are plenty of places where a `FieldName` of case `.foreign` would not be applicable (like in the `deleteFields` property). Would you like to lean into the type system here (not allowing "foreign field names" in places where they don't make sense) or rely on the programmer to know when it makes sense to specify a `.foreign` `FieldName`?

Reply:
I tend to lean toward flexibility in the actual data structures. Normally users will go through the API which is safer. But I could also see this making sense as a separate enum like `ForeignFieldName`.

Reply:
I think the separate `ForeignFieldName` option makes more sense to me in this particular case because the user is most likely (under the currently available API) to create `FieldConstraint`s directly (to pass into `SchemaBuilder`'s `field()` function as part of the `DatabaseSchema.FieldConstraint...` variadic list). Because of this, it is particularly easy to pass a non-foreign field name under the design where `foreign` is a case on `FieldName`.

Reply:
I gave that a go, let me know if you like how that shook out.